swift_kernel.py

```diff
@@ -426,6 +491,32 @@ def _process_install_line(self, line_index, line):
             'products': parsed[1:],
         }]
 
+    def _prepare_install_location(self, install_location):
+        swift_import_search_path = os.environ.get('SWIFT_IMPORT_SEARCH_PATH')
+        if swift_import_search_path is None:
+            raise PackageInstallException(
+                    'Install Error: Cannot install packages because '
+                    'SWIFT_IMPORT_SEARCH_PATH is not specified.')
+
+        if install_location is None:
+            os.makedirs(swift_import_search_path, exist_ok=True)
+            return
+
+        os.makedirs(install_location, exist_ok=True)
+
+        # symlink 'modules' to the user-specified location
+        # Remove existing 'modules' if it is already a symlink
+        try:
+            if stat.S_ISLNK(os.lstat(swift_import_search_path).st_mode):
+                os.unlink(swift_import_search_path)
+        except FileNotFoundError as e:
+            pass
+        except Error as e:
+            raise PackageInstallException(
+                    'Line %d: %s' % (line_index + 1, str(e)))
+
+        os.symlink(install_location, swift_import_search_path, target_is_directory=True)
+
     def _install_packages(self, packages, swiftpm_flags):
         if len(packages) == 0 and len(swiftpm_flags) == 0:
```

Comment:
When you implement my top-level comment, this should get an additional condition `install_location is None`.

Reply:
The condition was already there, unless I am misunderstanding what you mean.

Reply:
Ah yes, this is not relevant any more after the changes that you made.